llvm/tools/sycl-post-link/sycl-post-link.cpp

```diff
@@ -289,6 +291,42 @@ enum KernelMapEntryScope {
   Scope_Global     // single entry in the map for all kernels
 };
 
+static KernelMapEntryScope selectDeviceCodeSplitScopeAutomatically(Module &M) {
+  // Here we can employ various heuristics to decide which way to split kernels
```

Comment:
There are more comments then the code here :) I think the idea is pretty straightforward and the description could be shrinked.

Reply:
I've re-structured comments a bit in edfa44cc9a08c732d48f3bfe2b8caf3f8841d4ac